xarray/tests/test_dataarray.py

```diff
@@ -501,6 +501,18 @@ def test_getitem_dataarray(self):
         assert_equal(da[ind], da[[0, 1]])
         assert_equal(da[ind], da[ind.values])
 
+    def test_getitem_empty_index(self):
+        da = DataArray(np.arange(12).reshape((3, 4)), dims=['x', 'y'])
+        assert da[{'x': []}].size == 0
+        assert da[{'x': []}].x.size == 0
+        assert da[{'x': []}].y.size == da.y.size
```

Comment:
Instead of writing all these separate asserts, could you construct an expected DataArray and use xarray's `assert_identical()`? In addition to checking every DataArray property in one line of code, that also verifies additional internal invariants.